core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -486,18 +487,24 @@ private StartupConsistencyChecker(Long fileId) {
             inconsistentUris.add(parentUri);
           }
           for (Inode childInode : ((InodeDirectory) parentInode).getChildren()) {
-            AlluxioURI childUri = parentUri.join(childInode.getName());
-            if (childInode.isDirectory()) {
-              dirsToCheck.add(childInode.getId());
-            } else {
-              childInode.lockRead();
-              try {
+            try {
+              childInode.lockReadAndCheckParent(parentInode);
```

Comment:
could we return a Closeable from the different lock calls so that try-with-resource idiom can be used for releasing the lock?